src/main/scala/io/iohk/ethereum/jsonrpc/FilterManager.scala

```diff
@@ -0,0 +1,253 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.util.{ByteString, Timeout}
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.jsonrpc.EthService.BlockParam
+import io.iohk.ethereum.keystore.KeyStore
+import io.iohk.ethereum.ledger.BloomFilter
+import io.iohk.ethereum.transactions.PendingTransactionsManager
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+class FilterManager(
+    blockchain: Blockchain,
+    appStateStorage: AppStateStorage,
+    keyStore: KeyStore,
+    pendingTransactionsManager: ActorRef)
+  extends Actor {
+
+  import FilterManager._
+  import akka.pattern.{ask, pipe}
+
+  val maxBlockHashesChanges = 256
+
+  var filters: Seq[Filter] = Nil
+
+  var lastCheckBlocks: Map[BigInt, BigInt] = Map.empty
+
+  implicit val timeout = Timeout(5.seconds)
+
+  override def receive: Receive = {
+    case NewLogFilter(fromBlock, toBlock, address, topics) => addFilterAndSendResponse(LogFilter(generateId(), fromBlock, toBlock, address, topics))
+    case NewBlockFilter() => addFilterAndSendResponse(BlockFilter(generateId()))
+    case NewPendingTransactionFilter() => addFilterAndSendResponse(PendingTransactionFilter(generateId()))
+    case UninstallFilter(id) => uninstallFilter(id)
+    case GetFilterLogs(id) => getFilterLogs(id)
+    case GetFilterChanges(id) => getFilterChanges(id)
+  }
+
+  private def addFilterAndSendResponse(filter: Filter): Unit = {
+    filters :+= filter
+    lastCheckBlocks += (filter.id -> appStateStorage.getBestBlockNumber())
+    sender() ! NewFilterResponse(filter.id)
+  }
+
+  private def uninstallFilter(id: BigInt): Unit = {
```

Comment:
Within the definition it states: _Additonally Filters timeout when they aren't requested with eth_getFilterChanges for a period of time._  ... is that done? If so please point me out where is it and discard this comment :smile: